src/lazyexpression.jl

```diff
@@ -121,6 +124,12 @@ function optimize(expr::LazyExpression{typeof(*)}, ::Type{<:Union{Number, Variab
     LazyExpression(Functions.mul!, deepcopy(expr()), expr.args...)
 end
 
+function optimize(expr::LazyExpression{typeof(vcat)}, T::Type{<:Union{AbstractVector, Number}}...)
+    result = expr()
+    dest = similar(result)
+    dest .= result
```

Comment:
Let's just do `dest = deepcopy(expr())` to ensure that no data is shared between `result` and `dest`.

Reply:
I think deepcopy of an svector gives an svector, so we need similar(). I did intend to so similar(deepcopy(r)) but I guess I lost that line somewhere along the way.

Reply:
Hmm, OK.

For `matvecmul!` it's actually fine to use `SVector`; in fact I test for it:

https://github.com/tkoolen/SimpleQP.jl/blob/86e67cacf535213e14df327e0fddec208050abaf/test/lazyexpression.jl#L116

but that's because 1) `dest` is always a vector of `AffineFunction`, and 2) I don't call `setindex!` on the vector; instead I modify the existing elements.

The real application here is `vcat`ing `AbstractVector`s of `AffineFunction`s, so you could actually do the same thing as `matvecmul!` if you restrict the element type and modify already-allocated elements instead of assigning.

In fact, now that I think about it more, that's *necessary*, because otherwise you're sharing working buffers with other `LazyExpression`s.

Reply:
I should definitely document that last bit at some point, since it can lead to subtle bugs.